torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -47,7 +50,7 @@ void LinearImpl::pretty_print(std::ostream& stream) const {
 
 Tensor LinearImpl::forward(const Tensor& input) {
   AT_ASSERT(!options.with_bias() || bias.defined());
```

Comment:
We should remove this check to better match the Python version. :)

Reply:
We would need some other changes to `Linear` as well:
1. Change `LinearOptions` to take `in_features` / `out_features` / `bias` instead of `in` / `out` / `with_bias`, to better match the Python version. (We would need to fix pretty_print as well.)
2. When `options.bias()` is false, we should still register the `bias` parameter, to better match the Python version.
3. The rest of the `reset` code should strictly follow:
```python
        init.kaiming_uniform_(self.weight, a=math.sqrt(5))
        if self.bias is not None:
            fan_in, _ = init._calculate_fan_in_and_fan_out(self.weight)
            bound = 1 / math.sqrt(fan_in)
            init.uniform_(self.bias, -bound, bound)
```